### ISSUE TITLE
latency plugin needs love: rados bench output should be documented

### ISSUE THREAD

**Issue by grinapo** [grinapo/collectd-ceph#11]
Right now there's a regexp which tries to parse the bench output, which fails with my output.
I have patched it up in 5024baa, but they should be either unified or if not possible, thrown into an array and check in a loop. I (or someone nice) need to check older and newer output, and see if the regexp was buggy or the output have considerably changed.

**Comment by y4ns0l0** [comment]
Can you copy/paste your output ?

**Comment by Kallio** [comment]
I had a same issue and replacing the line 66 with just 
this  regex_match = re.compile('^([a-zA-Z]+) [lL]atency: +(\w+.?\w+)', re.MULTILINE)
fixed it without bigger modifications like https://github.com/grinapo/collectd-ceph/commit/5024baaecddd055d0a4cb16bc9008f08ed72f0c1

output with centos 7+ hammer 0.94.7 will at least give this:
```
 Maintaining 1 concurrent writes of 65536 bytes for up to 10 seconds or 0 objects
 Object prefix: benchmark_data_2467722
   sec Cur ops   started  finished  avg MB/s  cur MB/s  last lat   avg lat
     0       0         0         0         0         0         -         0
     1       1        66        65   4.06061    4.0625 0.0155893 0.0152715
     2       1       134       133   4.15496      4.25 0.0191927 0.0150341
     3       1       209       208   4.33224    4.6875  0.013498 0.0143977
     4       1       252       251   3.92099    2.6875  0.017076 0.0144669
     5       1       290       289   3.61172     2.375 0.0266884 0.0172712
     6       1       357       356   3.70758    4.1875 0.0150334 0.0168429
     7       1       422       421    3.7582    4.0625 0.0137151 0.0166162
     8       1       494       493   3.85084       4.50.00915379 0.0162211
     9       1       563       562   3.90205    4.3125 0.0156637 0.0160091
    10       1       633       632   3.94928     4.375 0.0169808 0.0158136
 Total time run:         10.022365
Total writes made:      634
Write size:             65536
Bandwidth (MB/sec):     3.954 

Stddev Bandwidth:       1.3996
Max bandwidth (MB/sec): 4.6875
Min bandwidth (MB/sec): 0
Average Latency:        0.0158045
Stddev Latency:         0.033237
Max latency:            0.844966
Min latency:            0.00689093
```

anyhow this rados bench in ceph code itself needs to be fixed. If your max bandwidth exceed 1GB/s the benchmark will divide it and result is 1MB/s. Also texts like Latency and latency...

**Comment by Kallio** [comment]
with ceph-0.94.9 the result is 
```
Maintaining 1 concurrent writes of 65536 bytes for up to 10 seconds or 0 objects
Object prefix: benchmark_data_2574018
  sec Cur ops   started  finished  avg MB/s  cur MB/s last lat(s)  avg lat(s)
    0       0         0         0         0         0           -           0
    1       1        66        65   4.06086    4.0625   0.0115033   0.0153253
    2       1       132       131   4.09256     4.125    0.015406   0.0151885
    3       1       201       200   4.16567    4.3125   0.0238941   0.0149759
    4       1       264       263   4.10837    3.9375   0.0140496   0.0151755
    5       1       333       332   4.14907    4.3125   0.0155221   0.0150548
    6       1       369       368   3.83253      2.25   0.0118395     0.01532
    7       1       369       368   3.28505         0           -     0.01532
    8       1       418       417   3.25715   1.53125   0.0154579   0.0185365
    9       1       441       440   3.05497    1.4375   0.0163541   0.0204252
   10       1       510       509   3.18066    4.3125   0.0155248   0.0196282
Total time run:         10.021715
Total writes made:      511
Write size:             65536
Bandwidth (MB/sec):     3.18683
Stddev Bandwidth:       1.75871
Max bandwidth (MB/sec): 4.3125
Min bandwidth (MB/sec): 0
Average IOPS:           50
Average Latency(s):     0.0196084
Stddev Latency(s):      0.0685118
Max latency(s):         1.37325
Min latency(s):         0.00716472
```

**Comment by y4ns0l0** [comment]
Here is a jewel ceph-10.2.3 output :

```
Maintaining 1 concurrent writes of 65536 bytes to objects of size 65536 for up to 10 seconds or 0 objects
Object prefix: benchmark_data_ylal8580_1014770
  sec Cur ops   started  finished  avg MB/s  cur MB/s last lat(s)  avg lat(s)
    0       0         0         0         0         0           -           0
    1       1         6         5  0.312469    0.3125    0.719529    0.190965
    2       1        16        15  0.468687     0.625    0.105129    0.128102
    3       1        22        21   0.43744     0.375   0.0309746    0.138013
    4       1        23        22  0.343705    0.0625    0.115548    0.136992
    5       1        23        22  0.274965         0           -    0.136992
    6       1        23        22  0.229136         0           -    0.136992
    7       1        25        24  0.214259 0.0416667    0.118578    0.273744
    8       1        28        27  0.210911    0.1875    0.306593    0.293086
    9       1        32        31  0.215251      0.25    0.662107    0.289549
   10       1        34        33  0.206224     0.125    0.108787    0.299365
   11       1        35        34  0.193157    0.0625    0.214815    0.296878
Total time run:         11.487340
Total writes made:      35
Write size:             65536
Object size:            65536
Bandwidth (MB/sec):     0.190427
Stddev Bandwidth:       0.193078
Max bandwidth (MB/sec): 0.625
Min bandwidth (MB/sec): 0
Average IOPS:           3
Stddev IOPS:            3
Max IOPS:               10
Min IOPS:               0
Average Latency(s):     0.328207
Stddev Latency(s):      0.623491
Max latency(s):         1.43747
Min latency(s):         0.00691456
Cleaning up (deleting benchmark objects)
Clean up completed and total clean up time :1.983606
```

**Comment by y4ns0l0** [comment]
should be fixed in #13 (  bench outputs above tested )

**Comment by grinapo** [comment]
Looks fine to me, merged. Thanks.
